src/IO/InputStreamFromROOTFile.cxx

```diff
@@ -76,13 +76,32 @@ InputStreamFromROOTFile::post_processing()
 
     stream_ptr->SetBranchAddress("time1", &time1);
     stream_ptr->SetBranchAddress("time2", &time2);
-
+    stream_ptr->SetBranchAddress("axialPos",&axialPos);
     stream_ptr->SetBranchAddress("eventID1",&eventID1);
     stream_ptr->SetBranchAddress("eventID2",&eventID2);
     stream_ptr->SetBranchAddress("energy1", &energy1);
     stream_ptr->SetBranchAddress("energy2", &energy2);
     stream_ptr->SetBranchAddress("comptonPhantom1", &comptonphantom1);
     stream_ptr->SetBranchAddress("comptonPhantom2", &comptonphantom2);
+    stream_ptr->SetBranchAddress("globalPosX1",&globalPosX1);
```

Comment:
why are we reading all these? can only take extra time. should it be optional?

Reply:
I can't remember what I said :-(

I think it's a useful feature but presumably would have to be optional. I don't know how ROOT works, but I can only imagine that it's going to slow down the reading. Also, it means that we'd handle only ROOT files where all these fields are present. (I have no clue if you can switch them off in GATE but I would think you could to compactify the ROOT file that way).

I'd be happy enough to comment it out for the moment, or introduce an extra bool to switch on/off the reading.

I

Reply:
I had a look at my comment

> InputStreamFromROOT constructors don't see all fields

I think what I meant that the constructor can be used to set various class variables, but not all (only `singles_readout_depth`?). As there are no `set_` functions, it means that the class can only be used via parsing. Not a major problem, but it's just "asymmetric".

Reply:
:ok: I created a set function and added the option to activate all fields